thriftbp/server_middlewares.go

```diff
@@ -17,6 +17,24 @@ var (
 	_ thrift.ProcessorMiddleware = ExtractDeadlineBudget
 )
 
+// BaseplateDefaultProcessorMiddlewares returns the default processor
+//  middlewares that should be used by a baseplate Thrift service.
+//
+// Currently they are (in order):
+//
+// 1. InjectServerSpan
+//
+// 2. InjectEdgeContext
+//
+// 3. ExtractDeadlineBudget
```

Comment:
I do think DeadlineBudget should be the first middleware on server and last on client (but on client all the additional ones will be after the default ones, so that's not possible, but it should at least be the last of the default ones). This is the relative time, so every extra in between will make it less accurate (e.g. if there's a middleware takes more than 1ms before it on the server or after it on the client, that is not take into consideration of the budget).

This won't be a problem for phase 2 as that's absolute time, but it is a problem for phase 1.

Reply:
Makes sense with one caveat, I think it needs to come _second_ on the server side after the Span step, reason being that Spans are how we monitor these requests, if we don’t set those up first, then requests could fail silently (from a server monitoring perspective)

Reply:
This one is resilient (won't fail or something), but as long as span creation one is fast enough I'm fine with that.